### ISSUE TITLE
Element picker wont open

### ISSUE THREAD

**Issue by Frejoh466** [gorhill/uBlock#3286]
### Describe the issue

Element picker wont open in Firefox 58.0b6 (64-bit) and uBlock origin 1.14.18. Works fine in 1.14.16.

### One or more specific URLs where the issue occurs

All sites but testing on www.Google.se

### Screenshot in which the issue can be seen

The buttons I'm pressing to try to enter element picker mode
https://i.imgur.com/rAhp7Rs.jpg

### Steps for anyone to reproduce the issue

1. Right click
2. Click on Block Element

or

1. Click on uBlock icon
2. Click on "Enter element picker mode"

### Your settings

Clean install of uBlock with default settings except enable advanced user. 

- OS/version: Microsoft Windows [Version 10.0.16299.64]
- Browser/version: Firefox 58.0b6 (64-bit) 
- uBlock Origin version: 1.14.18

##### Your filter lists

Default filter lists

##### Your custom filters (if any)

None.

##### Note

Does not work on Firefox 56, 57, or 58, does work in Chrome Version 62.0.3202.75 (Official Build) (64-bit). Tried to only have uBlock enabled. 

Not sure how to give you any more info as I found no debug log, and the logger shows nothing when I try to open the element picker.

**Comment by gorhill** [comment]
Unable to reproduce.

You will have to investigate on your side, there is something specific on your side causing this.

**Comment by gorhill** [comment]
Consider investigating your browser settings you may have changed, including your `userContent.css` file.

**Comment by gorhill** [comment]
Closing as per [CONTRIBUTING](https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md).

Read carefully 1st paragraph please.

**Comment by Frejoh466** [comment]
I apologies, I have read the CONTRIBUTING, multiple times. But I have no clue how to give you more information. The console gives nothing. I found how to open the debugger, but it doesn't throw me an error when I try to open the element picker. I do get errors, but nothing that is helpful.

`Unknown property ‘user-select’.  Declaration dropped. common.css:13:17`
When I hover over the uBlock icon.

`Unknown property ‘zoom’.  Declaration dropped.  dashboard.css:12:14
Unknown property ‘zoom’.  Declaration dropped.  dashboard-common.css:7:14`
When I open dashboard. 

`[uBlock0 cacheStorage] error { target: IDBRequest, isTrusted: true, currentTarget: IDBDatabase, eventPhase: 3, bubbles: true, cancelable: true, defaultPrevented: false, composed: false, timeStamp: 1383115.6325745347, cancelBubble: false, originalTarget: IDBRequest }  vapi-cachestorage.js:86:9
[uBlock0 cacheStorage] abort { target: IDBTransaction, isTrusted: true, currentTarget: IDBDatabase, eventPhase: 3, bubbles: true, cancelable: false, defaultPrevented: false, composed: false, timeStamp: 1383120.4832884383, cancelBubble: false, originalTarget: IDBTransaction }  vapi-cachestorage.js:86:9`
After I was idle for 30min.

`Error: Could not establish connection. Receiving end does not exist. undefined`
Also got that, not sure what this is.

But nothing when I try to open the element picker.

So it either doesn't send a request at all when I try to open the element picker, or the debugger doesn't show that action. This was a fresh install of windows 10, and is a fresh install with new profile of Firefox. So no changes has been made. uBlock 1.14.16 works, so that is a way to bypass this (it has other problems, but works as long as I don't clear the cache). 

I'm asking around where I can find better debug tools. I thought that the Firefox debug tool showed everything that the addon is doing, but that was my fault for assuming. I will report back once I find a tool that tells me what is happening.

**Comment by gorhill** [comment]
Does it work on other web page or is only `www.google.se` affected?

**Comment by gorhill** [comment]
Also, can you report how many cosmetic filters are in currently loaded in memory in the _3rd-party filters_ pane in the dashboard?

**Comment by gorhill** [comment]
Output of uBO's logger would be good also for when you load the page.

**Comment by Frejoh466** [comment]
It affect all sites I tried on, I have no way to open the element picker on 1.14.18

Currently I disabled every filter so I can easier debug.
"0 network filters ＋ 0 cosmetic filters from:"

logger,
```
05:58:06 |   |   |   | beacon | https://www.google.se/gen_204?atyp=i&ei=aUweWofLAaPv6QSstrKgDQ&vet=10ahUKEwjHo7aLjePXAhWjd5oKHSybDNQQsmQIDw..s&zx=1511935086354
-- | -- | -- | -- | -- | --
05:58:05 |   |   |   | script | https://apis.google.com/_/scs/abc-static/_/js/k=gapi.gapi.en.xcOUrevACUU.O/m=gapi_iframes,googleapis_client,plusone/rt=j/sv=1/d=1/ed=1/am=AAE/rs=AHpOoo_8YI-e6bhRV8OwDIX8yxw8HcW4bg/cb=gapi.loaded_0
05:58:05 |   |   |   | script | https://www.gstatic.com/og/_/js/k=og.og2.en_US.cWhXtkfAhl4.O/rt=j/m=def/exm=in,fot/d=1/ed=1/rs=AA2YrTuNdG7CKKW3KjCne5a2Yi7O_3r4bg
05:58:04 |   |   |   | image | https://adservice.google.se/adsid/google/ui
05:58:04 |   |   |   | beacon | https://www.google.se/gen_204?atyp=csi&ei=aUweWofLAaPv6QSstrKgDQ&s=webhp&imc=3&imn=3&imp=1&adh=&ima=1&ime=0&rt=aft.52,dcl.54,iml.160,ol.217,prt.52,xjs.1162,xjsee.1162,xjses.1098,xjsls.1046,wsrt.343,cst.119,dnst.0,rqst.191,rspt.85,sslt.47,rqstt.119,unt.0,cstt.0,dit.396&zx=1511935084436
05:58:04 |   |   |   | script | https://www.google.se/xjs/_/js/k=xjs.s.sv.OPANPLp7M8o.O/m=d3l,cbin,dgm,cbhb,aa,abd,async,dvl,foot,fpe,ipv6,lu,m,mu,sf,sonic/am=wCLJ8YIDIP8TQoQEVxAWlAUwGUA/exm=sx,c,sb,cdos,cr,elog,hsm,jsa,r,d,csi/rt=j/d=1/ed=1/t=zcms/rs=ACT90oEJxc12KZvMusnn_C8Nt46DilFn_g
05:58:04 |   |   |   | image | https://www.gstatic.com/inputtools/images/tia.png
05:58:04 |   |   |   | script | https://www.google.se/xjs/_/js/k=xjs.s.sv.OPANPLp7M8o.O/m=sx,c,sb,cdos,cr,elog,hsm,jsa,r,d,csi/am=wCLJ8YIDIP8TQoQEVxAWlAUwGUA/rt=j/d=1/t=zcms/rs=ACT90oEJxc12KZvMusnn_C8Nt46DilFn_g
05:58:03 |   |   |   | image | https://www.google.se/images/nav_logo242.png
05:58:03 |   |   |   | image | https://ssl.gstatic.com/gb/images/i2_2ec824b0.png
05:58:03 |   |   |   | image | https://www.google.se/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png
05:58:03 |   |   |   | image | https://www.google.se/images/branding/googlelogo/2x/googlelogo_color_120x44dp.png
05:58:03 |   |   |   | image | https://consent.google.com/status?continue=https://www.google.se&pc=s&timestamp=1511935081
05:58:03 |   |   |   | inline-script | https://www.google.se/
05:58:02 |   |   |   | doc | https://www.google.se/
  |   | https://www.google.se/


```

Version 1.14.16 can open the element picker, not with the right click menu, but with the "Enter element picker mode" button on the icon. It however also doesn't show anything happening in the debugging tool.

**Comment by gorhill** [comment]
> Currently I disabled every filter so I can easier debug.
> "0 network filters ＋ 0 cosmetic filters from:"

Well, that is your issue: #2963.

**Comment by Frejoh466** [comment]
I did not know the element picker get disabled when no filters are active. Shouldn't debug with filters of then. But even with filters on it dosn't work.

`142,087 network filters ＋ 82,467 cosmetic filters from:`

logger,
```
17:14:14 |   |   |   | script | https://apis.google.com/_/scs/abc-static/_/js/k=gapi.gapi.en.xcOUrevACUU.O/m=gapi_iframes,googleapis_client,plusone/rt=j/sv=1/d=1/ed=1/am=AAE/rs=AHpOoo_8YI-e6bhRV8OwDIX8yxw8HcW4bg/cb=gapi.loaded_0
-- | -- | -- | -- | -- | --
17:14:14 |   | \|\|google.*/gen_204?$other | -- | beacon | https://www.google.se/gen_204?atyp=i&ei=1NweWvGiF6rU6ASnyqDoCw&vet=10ahUKEwjx3N7oluTXAhUqKpoKHSclCL0QsmQIDw..s&zx=1511972054847
17:14:14 |   |   |   | script | https://www.gstatic.com/og/_/js/k=og.og2.en_US.hvBc4kgpQdA.O/rt=j/m=def/exm=in,fot/d=1/ed=1/rs=AA2YrTs_ngofTsE0VbwztD6RCCGxy0iQRg
17:14:13 |   | /adservice. | -- | image | https://adservice.google.se/adsid/google/ui
17:14:13 |   | \|\|google.*/gen_204?$other | -- | beacon | https://www.google.se/gen_204?atyp=csi&ei=1NweWvGiF6rU6ASnyqDoCw&s=webhp&imc=3&imn=3&imp=1&adh=&ima=1&ime=0&rt=aft.47,dcl.49,iml.48,ol.106,prt.47,xjs.1133,xjsee.1133,xjses.1071,xjsls.1028,wsrt.243,cst.0,dnst.0,rqst.167,rspt.65,sslt.0,rqstt.31,unt.0,cstt.0,dit.291&zx=1511972053980
17:14:13 |   |   |   | script | https://www.google.se/xjs/_/js/k=xjs.s.sv.OPANPLp7M8o.O/m=d3l,cbin,dgm,cbhb,aa,abd,async,dvl,foot,fpe,ipv6,lu,m,mu,sf,sonic/am=wCLJ8YIDIP8TQoQEVxAWlAUwGUA/exm=sx,c,sb,cdos,cr,elog,hsm,jsa,r,d,csi/rt=j/d=1/ed=1/t=zcms/rs=ACT90oEJxc12KZvMusnn_C8Nt46DilFn_g
17:14:13 |   |   |   | image | https://www.gstatic.com/inputtools/images/tia.png
17:14:13 |   |   |   | script | https://www.google.se/xjs/_/js/k=xjs.s.sv.OPANPLp7M8o.O/m=sx,c,sb,cdos,cr,elog,hsm,jsa,r,d,csi/am=wCLJ8YIDIP8TQoQEVxAWlAUwGUA/rt=j/d=1/t=zcms/rs=ACT90oEJxc12KZvMusnn_C8Nt46DilFn_g
17:14:12 |   |   |   | image | https://www.google.se/images/nav_logo242.png
17:14:12 |   | \|\|google.*/gen_204?$other | -- | beacon | https://www.google.se/gen_204?s=webaft&atyp=csi&ei=1NweWvGiF6rU6ASnyqDoCw&rt=wsrt.242,aft.47,prt.47
17:14:12 |   |   |   | image | https://ssl.gstatic.com/gb/images/i2_2ec824b0.png
17:14:12 |   |   |   | image | https://www.google.se/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png
17:14:12 |   |   |   | image | https://www.google.se/images/branding/googlelogo/2x/googlelogo_color_120x44dp.png
17:14:12 |   |   |   | image | https://consent.google.com/status?continue=https://www.google.se&pc=s&timestamp=1511972052
17:14:12 |   |   |   | inline-script | https://www.google.se/
17:14:12 |   |   |   | doc | https://www.google.se/
|   | https://www.google.se/
```

cosmetic filtering is on.

**Comment by gorhill** [comment]
You said in your opening comment that you use "Default filter lists" and no custom filters. If so, there is just no way you will end up with that many filters:

> 142,087 network filters ＋ 82,467 cosmetic filters from:

**Comment by Frejoh466** [comment]
Yes, that was with most filters on. With the default settings I have 
`66,494 network filters ＋ 44,227 cosmetic filters from:`

with the logger
```

19:17:56 |   | /adservice. | -- | image | https://adservice.google.se/adsid/google/ui
-- | -- | -- | -- | -- | --
19:17:56 |   | \|\|google.*/gen_204?$other | -- | beacon | https://www.google.se/gen_204?atyp=csi&ei=0PkeWoSdEOjJ6AS2novgBA&s=webhp&imc=3&imn=3&imp=3&adh=&ima=1&ime=0&rt=aft.71,dcl.48,iml.3198,ol.3691,prt.47,xjs.1356,xjsee.1356,xjses.1301,xjsls.1028,wsrt.279,cst.0,dnst.0,rqst.219,rspt.88,sslt.0,rqstt.37,unt.0,cstt.0,dit.326&zx=1511979476581
19:17:56 |   |   |   | image | https://www.google.se/images/nav_logo242.png
19:17:56 |   |   |   | script | https://apis.google.com/_/scs/abc-static/_/js/k=gapi.gapi.en.xcOUrevACUU.O/m=gapi_iframes,googleapis_client,plusone/rt=j/sv=1/d=1/ed=1/am=AAE/rs=AHpOoo_8YI-e6bhRV8OwDIX8yxw8HcW4bg/cb=gapi.loaded_0
19:17:54 |   | \|\|google.*/gen_204?$other | -- | beacon | https://www.google.se/gen_204?atyp=i&ei=0PkeWoSdEOjJ6AS2novgBA&vet=10ahUKEwiExei6suTXAhXoJJoKHTbPAkwQsmQIDw..s&zx=1511979474928
19:17:54 |   |   |   | script | https://www.google.se/xjs/_/js/k=xjs.s.sv.OPANPLp7M8o.O/m=d3l,cbin,dgm,cbhb,aa,abd,async,dvl,foot,fpe,ipv6,lu,m,mu,sf,sonic/am=wCLJ8YIDIP8TQoQEVxAWlAUwGUA/exm=sx,c,sb,cdos,cr,elog,hsm,jsa,r,d,csi/rt=j/d=1/ed=1/t=zcms/rs=ACT90oEJxc12KZvMusnn_C8Nt46DilFn_g
19:17:54 |   |   |   | image | https://www.gstatic.com/inputtools/images/tia.png
19:17:53 |   |   |   | script | https://www.gstatic.com/og/_/js/k=og.og2.en_US.hvBc4kgpQdA.O/rt=j/m=def/exm=in,fot/d=1/ed=1/rs=AA2YrTs_ngofTsE0VbwztD6RCCGxy0iQRg
19:17:53 |   |   |   | script | https://www.google.se/xjs/_/js/k=xjs.s.sv.OPANPLp7M8o.O/m=sx,c,sb,cdos,cr,elog,hsm,jsa,r,d,csi/am=wCLJ8YIDIP8TQoQEVxAWlAUwGUA/rt=j/d=1/t=zcms/rs=ACT90oEJxc12KZvMusnn_C8Nt46DilFn_g
19:17:52 |   | \|\|google.*/gen_204?$other | -- | beacon | https://www.google.se/gen_204?s=webaft&atyp=csi&ei=0PkeWoSdEOjJ6AS2novgBA&rt=wsrt.279,aft.72,prt.47
19:17:52 |   |   |   | image | https://ssl.gstatic.com/gb/images/i2_2ec824b0.png
19:17:52 |   |   |   | image | https://consent.google.com/status?continue=https://www.google.se&pc=s&timestamp=1511979472
19:17:52 |   |   |   | image | https://www.google.se/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png
19:17:52 |   |   |   | image | https://www.google.se/images/branding/googlelogo/2x/googlelogo_color_120x44dp.png
19:17:52 |   |   |   | inline-script | https://www.google.se/
19:17:52 |   |   |   | doc | https://www.google.se/
  |   | https://www.google.se/
```

**Comment by gorhill** [comment]
Open the browser console (ctrl-shift-j) and see if errors are reported when you try to launch the element picker.

**Comment by Frejoh466** [comment]
In ctrl + shift + k (console) with all logging on, it says nothing.

ctrl + shift + j (browser console) with all logging on, only gives me reflow when I press anything.
`reflow: 0.25ms`

**Comment by gorhill** [comment]
Is the picker `iframe` in the DOM after you try to launch it?

**Comment by Frejoh466** [comment]
So I read about DOM. If you are talking about the "Inspector" where the <!DOCTYPE html> is? If so then I did not find it. However, in version 1.14.16 where the element picker does work, I also didn't find it. Sorry, but I know nothing about web development.

I do believe that the element picker doesn't load properly. Can I manually try and force to load it?

**Comment by gorhill** [comment]
![a](https://user-images.githubusercontent.com/585534/33527357-2e3181ae-d81d-11e7-84ec-2069fa60c14b.png)

Element picker `iframe` is selected above.

**Comment by Frejoh466** [comment]
I do not have any `<iframe` tags. iframe is only mentioned inside `<script`

https://i.imgur.com/qyuGGtF.jpg

**Comment by gorhill** [comment]
Sorry, I don't know what else to say. Element picker seems to be working for everybody but you. At this point I am out of idea, you will need someone with dev skills looking at your own installation of Firefox.

**Comment by Frejoh466** [comment]
Ok, that's fine. 1.14.16 works so I can rollback whenever I want to use it. And as you said, seems like it's only I who have this problem.

But I do appreciate the help, so thank you. And thanks for creating this addon.

**Comment by gorhill** [comment]
Note that I don't have FF58, I test on FF57 or FF59. Given that it works on both on my side, I am assuming there is no reason it should not work on FF58. I might give it a try just in case.

**Comment by Frejoh466** [comment]
Ok, I found a solution. I don't know why, but it works.

These filters cannot be active.

- Adguard Mobile Filters
- Fanboy’s Annoyance List​
- Fanboy’s Anti-Thirdparty Social (see warning inside list)​
- Fanboy’s Cookiemonster List​
- Fanboy’s Social Blocking List​
- Dan Pollock’s hosts file​
- hpHosts’ Ad and tracking servers​
- MVPS HOSTS

If I ever enable any of the said filters, I must disable it, "Purge all caches", then delete the current profile Firefox cache. And it should work. If I install fresh UBO i need to do the same, but the filter should already be deactivated, so just purge the cache and delete the Firefox cache.

There is nothing inside the filters that break the element picker. I opened them and added the rules to "My Filters", and the element picker still works. But I don't know why it breaks by activate them.
